### PR TITLE
CI: Re-enable Windows for testing

### LINKED ISSUE #2: CI fails on Windows
Currently our tests are failing on Windows, although I am pretty sure that we don't have any Linux specific stuff in there. Thus we need to fix that at some point.

### PR DESCRIPTION
**Pull request by SchoolGuy**
Fixes #2 

The scope of this PR was expanded due to details that appeared during the actual bugfix. This PR now contains changes for the following things:

- CI: Enable Windows for testing
- CI: Only send coverage to Codacy when running on Linux
- Tests: Replace https://github.com/ContainerSolutions/go-utils with our own code
- Tests: Add support for multi-buffered requests and responses
- Items: `enable_gpxe` was renamed to `enable_ipxe`
- Client: Add helper method `IsValueInherit`
- Tests: Remove mock system creation and use the real method

The code in this PR was not validated against a running Cobbler instance.

### CODE REVIEW

**Comment by SchoolGuy** [comment]
It appears that we are running into https://github.com/golang/go/issues/51269

**Comment by SchoolGuy** [comment]
After a bit of investigation, it appears that we are using the `StubHTTPClient` which is coming from <http://github.com/ContainerSolutions/go-utils>. This is a problem as the comparison that lets the Windows CI fail is done in that package.

**Comment by SchoolGuy** [comment]
I will extend the scope of this PR to also remove go-utils from the dependencies of the project so we can have our own implementation of an HTTPClient. This allows us then to use buffered responses for multiple requests and responses per-test.

**Comment by SchoolGuy** [comment]
Working on this PR showed that the creation of objects is not possible since we are using the embedded item structs due to the method `updateCobblerFields` not being able to handle that.

**Comment by SchoolGuy** [comment]
Okay Windows likes use now. Let's get back to fixing the skipped tests.

File: authentication_test.go [hunk 1]
```diff
@@ -53,47 +51,47 @@ func TestLogin(t *testing.T) {
...
+	c := createStubHTTPClientSingle(t, "login-err")
 	expected := `Fault(1): <class 'cobbler.cexceptions.CX'>:'login failed (cobbler)'`
 
 	ok, err := c.Login()
 	if ok {
 		t.Errorf("false expected; got true")
 	}
 
-	if err.Error() != expected {
+	if err != nil && err.Error() != expected {
```
Comment (tiltingpenguin) [inline authentication_test.go]: Is there a reason to test for err != nil here? In the theoretical case that we get an empty ok and an empty error this test would pass

File: cobblerclient.go [hunk 2]
```diff
@@ -357,33 +371,28 @@ func (c *Client) updateCobblerFields(what string, item reflect.Value, id string)
...
-		var value interface{}
-		switch v.Type().String() {
-		case "string", "bool", "int64", "int":
-			value = v.Interface()
-		case "[]string":
-			value = strings.Join(v.Interface().([]string), " ")
-		}
-		if result, err := c.Call(method, id, field, value, c.Token); err != nil {
+
+		if result, err := c.Call(method, id, field, v.Interface(), c.Token); err != nil {
```
Comment (tiltingpenguin) [inline cobblerclient.go:386]: There seems to be a missing newline or is this some quirk in go format?

**Review by tiltingpenguin — COMMENTED** [review]
I just have some minor things I want to clarify, otherwise LGTM. One general comment I want to make though is that the DeleteItemRecursive tests seem kind of useless to me since we only test them in the false case where the behaviour is the same as in the non recurisve function

**Review by tiltingpenguin — APPROVED** [review]

**Pull request merged**